### PR TITLE
[ML] adds new mpnet tokenization for nlp models

### PR DESCRIPTION
**Pull request by benwtrent**
This commit adds support for MPNet based models.

MPNet models differ from BERT style models in that:

 - Special tokens are different
 - Input to the model doesn't require token positions.

To configure an MPNet tokenizer for your pytorch MPNet based model:

```
"tokenization": {
  "mpnet": {...}
}
```
The options provided to `mpnet` are the same as the previously supported `bert` configuration.

### CODE REVIEW

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/nlp/tokenizers/BertTokenizer.java [hunk 1]
```diff
@@ -63,37 +69,99 @@ protected BertTokenizer(
...
+        this.bosToken = bosToken;
+        this.sepToken = sepToken;
+        this.clsToken = clsToken;
+        this.padToken = padToken;
+        this.maskToken = maskToken;
+        this.unknownToken = unknownToken;
+    }
+
+    public String getBosToken() {
+        return bosToken;
```
Comment (davidkyle) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/nlp/tokenizers/BertTokenizer.java:143]: I can't see BOS being used anywhere and there isn't a EOS (end of sequence). Is this a leftover?

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/nlp/tokenizers/BertTokenizer.java [hunk 2]
```diff
@@ -251,7 +319,7 @@ public TokenizationResult buildTokenizationResult(List<TokenizationResult.Tokeni
             tokenMap[i] = tokenPositionMapSeq1.get(i - 1);
             i++;
         }
-        tokenIds[i] = vocab.get(SEPARATOR_TOKEN);
+        tokenIds[i] = vocab.get(sepToken);
```
Comment (davidkyle) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/nlp/tokenizers/BertTokenizer.java:322]: The huggingface docs suggest that 2 separator tokens should be used for a pair of input sequences:  `pair of sequences: <s> A </s></s> B </s>` 

https://huggingface.co/docs/transformers/model_doc/mpnet#transformers.MPNetTokenizer.build_inputs_with_special_tokens

**Review by davidkyle — COMMENTED** [review]
Yay another supported model!

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/nlp/tokenizers/BertTokenizer.java (hunk 1 above)
Comment (benwtrent) [reply x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/nlp/tokenizers/BertTokenizer.java:143]: MPNet has this concept of "bos" I can remove it. From what I can tell, its almost always the same as the class token "s"

**Review by davidkyle — APPROVED** [review]
LGTM

**Pull request merged**